src/tests/ie_test_utils/common_test_utils/postgres_helpers.cpp

```diff
@@ -0,0 +1,584 @@
+        winVersion << " Datacenter Edition";
+    if (osVersionInfo.wSuiteMask & VER_SUITE_ENTERPRISE)
+        winVersion << " Enterprise Edition";
+    if (osVersionInfo.wSuiteMask & VER_SUITE_EMBEDDEDNT)
+        winVersion << " Embedded";
+    if (osVersionInfo.wSuiteMask & VER_SUITE_PERSONAL)
+        winVersion << " Home";
+    if (osVersionInfo.wSuiteMask & VER_SUITE_SMALLBUSINESS)
+        winVersion << " Small Business";
+
+    winVersion << " Build: " << osVersionInfo.dwBuildNumber;
+
+    return winVersion.str();
+#endif
+}
+
+std::string get_executable_name(void) {
+#ifdef _WIN32
+    char cFilePath[MAX_PATH] = {};
+    GetModuleFileName(nullptr, cFilePath, MAX_PATH);
+    std::string filePath(cFilePath);
+#else
+    std::string filePath;
+    std::ifstream("/proc/self/comm") >> filePath;
+    return filePath;
+#endif
+    return filePath.substr(filePath.find_last_of("/\\") + 1);
+}
+
+std::string get_hostname(void) {
+#ifdef _WIN32
+    DWORD szHostName = MAX_COMPUTERNAME_LENGTH;
+    char cHostName[MAX_COMPUTERNAME_LENGTH + 1] = {};
+    if (FAILED(GetComputerName(cHostName, &szHostName))) {
+        std::cerr << PG_ERR << "Cannot get a host name\n";
+#else
+    char cHostName[HOST_NAME_MAX];
+    if (gethostname(cHostName, HOST_NAME_MAX)) {
+        std::cerr << PG_ERR << "Cannot get a host name\n";
+        return "NOT_FOUND";
+#endif
+    }
+    return cHostName;
+}
+
+void add_pair(std::map<std::string, std::string>& keyValues, const std::string& key, const std::string& value) {
+    size_t dPos;
+    // Parse IR_name for opName and hash
+    if (key == "IR_name" && (dPos = value.find('_')) != std::string::npos) {
+        keyValues["opName"] = value.substr(0, dPos);
+        keyValues["opSet"] = "unknown";  // Need to set
+        keyValues["hashXml"] = value.substr(dPos + 1);
+        keyValues["pathXml"] = value;
+        return;
+    }
+    if (key == "Op") {
+        if ((dPos = value.find('.')) == std::string::npos) {
+            keyValues["opName"] = value.substr(0, dPos);
+            keyValues["opSet"] = "unknown";  // Need to set later
+        } else {
+            keyValues["opName"] = value.substr(0, dPos);
+            keyValues["opSet"] = value.substr(dPos + 1);
+        }
+    }
+    // Parse IR for opName and hash
+    if (key == "IR") {
+        keyValues["hashXml"] = value;
+        keyValues["pathXml"] = value + ".xml";
+        return;
+    }
+    // Parse Function for opName and opSet
+    if (key == "Function" && (dPos = value.find('_')) != std::string::npos) {
+        keyValues["opName"] = value.substr(0, dPos);
+        keyValues["opSet"] = value.substr(dPos + 6);  // Skipping "opset"
+        return;
+    }
+    // Normalize target devices
+    if (key == "target_device" || key == "TargetDevice" || key == "Device" || key == "targetDevice") {
+        if (value == "CPU") {
+// see https://sourceforge.net/p/predef/wiki/Architectures/
+#if defined(__arm__) || defined(_M_ARM) || defined(__ARMEL__)
```

Comment:
refer to comment about devices

Reply:
I don't have access to the Core from here to ask anything. Strongly recommend to do not change it to usage OpenVINO Core because it is a common implementation. I may suggest only an alignment names between Core results and here.

Reply:
We can create the core,as we enter the app, because we have device name. Initialisation of test env is covered by 'run_all_tests', which are defined in the end of entry point function

Anyway, if you have any concerns, let's discuss offline to avoid misunderstanding

Reply:
It is used for another reasons.